src/datatypes/src/value.rs

```diff
@@ -654,6 +655,16 @@ impl ListValue {
             Arc::new(new_item_field(output_type.item_type().as_arrow_type())),
         ))
     }
+
+    fn byte_size(&self) -> usize {
+        let mut size = size_of::<ConcreteDataType>();
+        if let Some(items) = self.items() {
+            for item in items.iter() {
+                size += item.as_value_ref().byte_size();
+            }
+        }
+        size
+    }
```

Comment:
I think we only need to consider the size of all items. We can use `'the size of the first item' * 'length of items'` to estimate the size.

Reply:
for ListValue, if it construct from `vec![None, Some(vec![Some(1), Some(2), Some(3)]), Some(vec![Some(4), None, Some(6)])];` , the size of the first item is `None` , and the size of all items is 0.

Reply:
this method is very fast, but it could be not perform well in some cases, for example, if the first item is `None`, it will return 0 or there are many `None` in items.